pkg/nebula/meta_client.go

```diff
@@ -234,27 +237,65 @@ func (m *metaClient) GetLeaderCount(leaderHost string) (int, error) {
 }
 
 func (m *metaClient) BalanceLeader() error {
+	log := getLog()
+	log.Info("balance leader")
 	req := &meta.LeaderBalanceReq{}
 	resp, err := m.client.LeaderBalance(req)
 	if err != nil {
 		return err
 	}
 	if resp.Code != meta.ErrorCode_SUCCEEDED {
+		if resp.Code == meta.ErrorCode_E_LEADER_CHANGED {
+			log.Info("request leader changed", "host", resp.Leader.Host, "port", resp.Leader.Port)
+			leader := fmt.Sprintf("%v:%v", resp.Leader.Host, resp.Leader.Port)
+			if err := m.updateClient(leader); err != nil {
+				return errors.Errorf("update client failed: %v", err)
+			}
+			resp, err := m.client.LeaderBalance(req)
+			if err != nil {
+				return err
+			}
+			if resp.Code != meta.ErrorCode_SUCCEEDED {
+				return errors.Errorf("retry balance leader code %d", resp.Code)
+			}
+		} else if resp.Code == meta.ErrorCode_E_BALANCED {
+			log.Info("cluster is balanced")
+			return nil
+		}
 		return errors.Errorf("BalanceLeader code %d", resp.Code)
 	}
 	return nil
 }
 
+func (m *metaClient) IsBalanced(hostItem []*meta.HostItem) bool {
```

Comment:
How to ensure that the algorithm here is consistent with `nebula`'s?

Reply:
Is it leader , data or both balanced? Does the function name need to be adjusted?